client/extensions/woocommerce/app/settings/shipping/shipping-zone/index.js

```diff
@@ -49,18 +53,35 @@ class Shipping extends Component {
 	}
 
 	render() {
-		const { siteId, className, loaded, markSaved, markChanged, params } = this.props;
+		const { siteId, className, loaded, markSaved, markChanged, params, zone, locations, actions, translate } = this.props;
 		const isRestOfTheWorld = 0 === Number( params.zone );
 
+		const onSave = () => {
+			actions.changeShippingZoneName( siteId, getZoneName( zone, locations, translate ) );
```

Comment:
No need to do this. If you submit a zone with an empty name, WooCommerce core autogenerates its name ([here](https://github.com/woocommerce/woocommerce/blob/513bca9fa54e3259f2127af416774e57e285f8b8/includes/class-wc-shipping-zone.php#L107)).